### PR TITLE
Create Announcements Page

### LINKED ISSUE #24: Create Create Announcements Page
EPIC: Frontend for Announcements on Play Page 
Issue F2
Create "Create Announcements Page"

### PR DESCRIPTION
**Pull request by yixuan-wong**
## Overview
Created "Create Announcements Page" and updated page header at ```/admin/announcements/{commonsId}/create``` to show the name of the commons.

## Screenshots 
image

## Tests
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Dokku Dev Link
https://happycows-dev-yixuan-wong.dokku-12.cs.ucsb.edu

## Validation 
1. Log in with valid account 
2. Under "Admin" in the top bar click on "List Commons"
3. Scroll to the far right column on the table 
4. Click on "Announcements"
5. Click on "Create Announcements"

## Linked Issues
Closes #24

### CODE REVIEW

**Review by yungongWang — APPROVED** [review]
LGTM! I log in the website with the validation step, everything looks great! The Announcement page works well.

**Review by yungongWang — DISMISSED** [review]
LGTM! I followed the validation step, and everything works really great. The announcement is successfully created.

**Review by pconrad — CHANGES_REQUESTED** [review]
Can we replace this unwieldy free form text field that requires entering dates in ISO format (which is not ideal), with this nice compact representation from the Create Commons Form?

image

**Review by yungongWang — APPROVED** [review]
LGTM! The date is transformed to nicer format, and the frontend is fixed

File: frontend/src/main/components/Announcement/AnnouncementForm.js [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
-    // Stryker disable next-line all
-    //const yyyyq_regex = /((19)|(20))\d{2}[1-4]/i; // Accepts from 1900-2099 followed by 1-4.  Close enough.
+    // Stryker disable all
+    const onSubmit = (data) => {
+        if (!data.endDate) {
+            data.endDate = null;
+        }
+        submitAction(data);
+    };
+    // Stryker restore all
```
Comment (pconrad) [inline frontend/src/main/components/Announcement/AnnouncementForm.js:50]: A stryker exception here is a bad idea.  This is the most important code to test!

Let's remove the stryker exception and write the necessary tests.

File: frontend/src/main/components/Announcement/AnnouncementForm.js [hunk 2]
```diff
@@ -1,41 +1,61 @@
...
+    } = useForm({
+        defaultValues: {
+            startDate: initialContents?.startDate 
+                ? initialContents.startDate.split("T")[0]
+                : getPSTDate(),
+            endDate: initialContents?.endDate || null,
+            announcementText: initialContents?.announcementText || "",
+            ...initialContents, 
+        },
+    });
```
Comment (pconrad) [inline frontend/src/main/components/Announcement/AnnouncementForm.js:20]:
```suggestion
    });
    // Stryker restore all
```

You can't just turn off Stryker whenever you feel like it.  There are only very narrow cases where we are permitted Stryker exceptions, and even some of those should actually be removed from the code base as we know *do* know how to test those.  The number of Stryker exceptions needs to be going *down* not *up*.

File: frontend/src/main/components/Announcement/AnnouncementForm.js [hunk 3]
```diff
@@ -1,41 +1,61 @@
...
+            day: '2-digit',
+        };
+        const formatter = new Intl.DateTimeFormat('en-US', options);
+        const [month, day, year] = formatter.formatToParts(now)
+            .filter(({ type }) => type === 'month' || type === 'day' || type === 'year')
+            .map(({ value }) => value);
+
+        return `${year}-${month}-${day}`;
+    }
+    // Stryker restore all
```
Comment (pconrad) [inline frontend/src/main/components/Announcement/AnnouncementForm.js:41]: You absolutely should be testing this code.

File: frontend/src/main/components/Announcement/AnnouncementForm.js [hunk 4]
```diff
@@ -1,41 +1,61 @@
...
 
-    // Stryker disable next-line all
-    //const yyyyq_regex = /((19)|(20))\d{2}[1-4]/i; // Accepts from 1900-2099 followed by 1-4.  Close enough.
+    // Stryker disable all
```
Comment (pconrad) [inline frontend/src/main/components/Announcement/AnnouncementForm.js:43]:
```suggestion

```

File: frontend/src/main/components/Announcement/AnnouncementForm.js (hunk 1 above)
Comment (pconrad) [inline frontend/src/main/components/Announcement/AnnouncementForm.js:50]:
```suggestion

```

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the stryker exceptions noted and leave only the one around the useForm call. That's the only one that's approved.

For the rest of the code, you need to write the tests.

**Pull request closed without merging**